src/spec-configuration/containerFeaturesOrder.ts

```diff
@@ -3,133 +3,538 @@
+				return aTag.localeCompare(bTag);
+			}
+
+			// Sort by options
+			// TODO: Compares options Record by stringifying them.  This isn't fully correct.
+			if (aOptions !== bOptions) {
+				return aOptions.localeCompare(bOptions);
+			}
+
+			// Sort by manifest digest hash
+			if (aDigest !== bDigest) {
+				return aDigest.localeCompare(bDigest);
+			}
+
+			// Consider these two OCI Features equal.
+			return 0;
+
+		case 'direct-tarball':
+			throw new Error(`Feature type 'direct-tarball' will be supported but is unimplemented.`);
+
+		case 'file-path':
+			bSourceInfo = bSourceInfo as FilePathSourceInformation;
+			return aSourceInfo.resolvedFilePath.localeCompare(bSourceInfo.resolvedFilePath);
+
+		default:
+			throw new Error(`Feature dependencies are only supported in Features published (2) to an OCI registry, (2) as direct HTTPS tarball, or (3) as a local Feature.  Got type: '${aSourceInfo.type}'.`);
+	}
+}
+
+function featureSupportsDependencies(featureSet?: FeatureSet): boolean {
+	if (!featureSet) {
+		return false;
+	}
+
+	const publishType = featureSet.sourceInformation.type;
+	// TODO: Implement 'direct-tarball'.
+	return publishType === 'oci' /*|| publishType === 'direct-tarball'*/ || publishType === 'file-path';
+}
+
+async function applyOverrideFeatureInstallOrder(
+	params: CommonParams,
+	processFeature: (userFeature: DevContainerFeature) => Promise<FeatureSet | undefined>,
+	worklist: FNode[],
+	config: { overrideFeatureInstallOrder?: string[] },
+) {
+	const { output } = params;
+
+	if (!config.overrideFeatureInstallOrder) {
+		return worklist;
+	}
+
+	const overrideFeatureNodes: FNode[] = [];
+
+	// Create an artificial node for each Feature in the override property.
+	// Reverse iterate to remove out items from config that have been processed.
+	for (let j = config.overrideFeatureInstallOrder.length - 1; j >= 0; j--) {
+		const userFeatureId = config.overrideFeatureInstallOrder[j];
+		const overrideNode: FNode = {
+			type: 'artificial',
+			userFeatureId,
+			options: {},
+			dependsOn: [],
+			installsAfter: [],
+		};
+		const processed = await processFeature(overrideNode);
+		if (!processed) {
+			throw new Error(`Feature '${userFeatureId}' in 'overrideFeatureInstallOrder' could not be processed.`);
+		}
+
+		if (featureSupportsDependencies(processed)) {
+			overrideNode.featureSet = processed;
+			overrideFeatureNodes.push(overrideNode);
+			config.overrideFeatureInstallOrder.splice(j, 1);
+		}
+	}
+
+	overrideFeatureNodes.reverse();
+	output.write(`[override-feature-nodes]: ${overrideFeatureNodes.map(n => n.userFeatureId).join(', ')}`, LogLevel.Info);
+
+	for (let i = 0; i < overrideFeatureNodes.length - 1; i++) {
+		// Scan the resolvedGraph for a node that matches the next override node.
+		const currentOverrideNode = overrideFeatureNodes[i];
+		const nextOverrideNode = overrideFeatureNodes[i + 1];
+		let countMatches = 0;
+		for (const node of worklist) {
+			if (satisfiesSoftDependency(node, nextOverrideNode)) {
```

Comment:
Somehow we're not looking at nodes with `satisfiesSoftDependency(node, overrideFeatureNodes[0])`?

Reply:
I guess we need another loop looking at `satisfiesSoftDependency(node, currentOverrideNode)` and adding `currentOverrideNode.installsAfter.push(node)`?

Reply:
The goal of this is to implement the suggestion here https://github.com/devcontainers/spec/pull/234#discussion_r1183470261

Reply:
This has been removed in favor of a priority-based approach evaluated per-round. See https://github.com/devcontainers/cli/pull/530/commits/c8bfd1b40c50603da78d9066c80f10e3669d158c